langchain/llms/sagemaker_endpoint.py

```diff
@@ -0,0 +1,132 @@
+            se = SagemakerEndpoint(
+                endpoint_name=endpoint_name,
+                role_arn="role_arn",
+                role_session_name="role_session_name"
+            )
+    """
+
+    endpoint_name: str = ""
+    """# The name of the endpoint. Must be unique within an AWS Region."""
+    task: Optional[str] = None
+    """Task to call the model with. Should be a task that returns `generated_text`."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    role_arn: Optional[str] = None
+    role_session_name: Optional[str] = None
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        _model_kwargs = self.model_kwargs or {}
+        return {
+            **{"endpoint_name": self.endpoint_name, "task": self.task},
+            **{"model_kwargs": _model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "sagemaker_endpoint"
+
+    def _call(self, prompt: str, stop: Optional[List[str]] = None) -> str:
+        """Call out to Sagemaker inference endpoint.
+
+        Args:
+            prompt: The prompt to pass into the model.
+            stop: Optional list of stop words to use when generating.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+                response = se("Tell me a joke.")
+        """
+        import boto3
+
+        session = boto3.Session(profile_name="test-profile-name")
+        sagemaker_runtime = session.client("sagemaker-runtime", region_name="us-west-2")
+
+        # TODO: use AWS IAM assumed roles to authenticate from the EC2 instance
+        # def role_arn_to_session(**args):
+        #     """
+        #     Usage :
+        #         session = role_arn_to_session(
+        #             RoleArn='arn:aws:iam::012345678901:role/example-role',
+        #             RoleSessionName='ExampleSessionName')
+        #         client = session.client('sqs')
+        #     """
+        #     client = boto3.client('sts')
+        #     response = client.assume_role(**args)
+        #     return boto3.Session(
+        #         aws_access_key_id=response['Credentials']['AccessKeyId'],
+        #         aws_secret_access_key=response['Credentials']['SecretAccessKey'],
+        #         aws_session_token=response['Credentials']['SessionToken'])
+
+        # session = role_arn_to_session(RoleArn="$role-arn",
+        # RoleSessionName="test-role-session-name")
+        # sagemaker_runtime = session.client(
+        # "sagemaker-runtime", region_name="us-west-2"
+        # )
+
+        _model_kwargs = self.model_kwargs or {}
+
+        # payload samples
+        parameter_payload = {"inputs": prompt, "parameters": _model_kwargs}
```

Comment:
This is assuming a certain input format from the model endpoint which might not always be the case. A better approach might be to provide a function as input to this class to transform the inputs to LLM to match with what the model will expect.